### PR TITLE
rocksdb 3.12

### PR DESCRIPTION
**Pull request by dch**

### CODE REVIEW

**Comment by dch** [comment]
- bump version
- tighten brew audit compliance

**Comment by mikemcquaid** [comment]
Audit still failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/30035/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_rocksdb/

**Comment by dch** [comment]
@mikemcquaid thanks I was not aware that brew audit can be run post build...

```
-r--r--r--  1 dch  admin   5.1M 22 Jul 19:38 librocksdb.a
lrwxr-xr-x  1 dch  admin    23B 22 Jul 19:38 librocksdb.dylib@ -> librocksdb.dylib.3.12.0
lrwxr-xr-x  1 dch  admin    23B 22 Jul 19:38 librocksdb.dylib.3@ -> librocksdb.dylib.3.12.0
lrwxr-xr-x  1 dch  admin    23B 22 Jul 19:38 librocksdb.dylib.3.12@ -> librocksdb.dylib.3.12.0
-r--r--r--  1 dch  admin   2.5M 22 Jul 19:38 librocksdb.dylib.3.12.0
```

So these are clearly all library related but this looks like one *could* remove the symlinks in the recipe. Do you think that might break anything? The alternative is that `brew audit` is less pedantic.

**Comment by dch** [comment]
So here's a patch, that takes care of this, and AFAICT doesn't break `brew test` for rocksdb and osquery, the only 2 dependencies.

**Comment by dch** [comment]
FWIW submitted https://github.com/facebook/rocksdb/issues/666 upstream muahhahahaha

**Comment by dch** [comment]
upstream PR for a permanent fix https://github.com/facebook/rocksdb/pull/670 so we can take this out in the next rocksdb release.

File: Library/Formula/rocksdb.rb [hunk 1]
```diff
@@ -20,6 +18,9 @@ class Rocksdb < Formula
   depends_on "snappy"
   depends_on "lz4"
 
+  # fix the weird library naming scheme in rocksdb Makefile
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/rocksdb.rb]: Mind using https://github.com/facebook/rocksdb/pull/670.patch ?

Comment (dch) [inline Library/Formula/rocksdb.rb]: done with thanks!

Comment (DomT4) [inline Library/Formula/rocksdb.rb]: Doesn't seem to be done?

Comment (dch) [inline Library/Formula/rocksdb.rb]: On Fri, 31 Jul 2015, at 04:56 PM, Dominyk Tiller wrote:
> > @@ -20,6 +18,9 @@ class Rocksdb < Formula
> >    depends_on "snappy"
> >    depends_on "lz4"
> >  
> > +  # fix the weird library naming scheme in rocksdb Makefile
> > +  patch :DATA
> 
> Doesn't seem to be done?

It has been updated, yes. The :DATA section matches
https://patch-diff.githubusercontent.com/raw/facebook/rocksdb/pull/670.patch
if you look closely. It's not that different from my original patch
though. The main difference is platform detection & commit details.

Comment (mikemcquaid) [inline Library/Formula/rocksdb.rb]: Sorry, I mean use `https://github.com/facebook/rocksdb/pull/670.patch` as a `patch do` block (unless it doesn't apply cleanly). Either way it'd be good to have that linked in a comment so there's more context. Thanks!

**Comment by yhchiang** [comment]
We are cutting RocksDB release 3.13 this week, and the pull request @dch submitted has been merged to rocksdb master.
https://github.com/facebook/rocksdb/pull/670

**Comment by bfontaine** [comment]
This PR should be rebased on top of `master`.

**Comment by dch** [comment]
We can close this, patch is upstreamed now. Thanks @mikemcquaid and team as usual for your guidance & tips!

**Pull request closed without merging**